main.go

```diff
@@ -0,0 +1,230 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2/textlogger"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	log    = ctrl.Log.WithName("spectro-cleanup")
+
+	// optional env vars to override default configuration
+	cleanupSeconds     int64
+	propagationPolicy  = metav1.DeletePropagationBackground
+	cleanupSecondsStr  = os.Getenv("CLEANUP_DELAY_SECONDS")
+	fileConfigPath     = os.Getenv("CLEANUP_FILE_CONFIG_PATH")
+	resourceConfigPath = os.Getenv("CLEANUP_RESOURCE_CONFIG_PATH")
+	saName             = os.Getenv("CLEANUP_SA_NAME")
+	roleName           = os.Getenv("CLEANUP_ROLE_NAME")
+	roleBindingName    = os.Getenv("CLEANUP_ROLEBINDING_NAME")
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	initConfig()
+}
+
+type DeleteObj struct {
+	schema.GroupVersionResource
+	Name      string
+	Namespace string
+}
+
+func main() {
+	ctrl.SetLogger(textlogger.NewLogger(textlogger.NewConfig()))
```

Comment:
this line replaces the following to address a deprecation warning that was causing lint errors:
```
	ctrl.SetLogger(klogr.New())

```